src/parser.js

```diff
@@ -689,9 +689,13 @@ export class GenericParser extends Tokenizer {
 
   parseForStatement() {
     this.lex();
+    let isAwait = this.eat(TokenType.AWAIT);
```

Comment:
It looks like this would allow `for await` even outside of async functions, which is not legal. Can you ensure that's not the case and add a test to that effect?